File: examples/equos-nodejs-integration/src/create-agent.ts
import "dotenv/config";

import {
  Equos,
  AgentProvider,
  type EquosAgent,
  GeminiRealtimeModels,
  GeminiRealtimeVoices,
} from "@equos/node-sdk";

const client = Equos.client(process.env.EQUOS_API_KEY!);

export const createAgent = async () => {
  const agent: EquosAgent = await client.agents.create({
    name: "French Translator",
    provider: AgentProvider.gemini,
    instructions:
      "You are a helpful assistant that translates English to French.",
    model: GeminiRealtimeModels.gemini_2_5_flash_native_audio_09_2025,
    voice: GeminiRealtimeVoices.Charon,
  });

  console.log("Agent created:", agent.id);

  console.log("Agent details:");
  console.log(agent);
};

createAgent();

File: examples/equos-python-integration/src/python_integration/create_agent.py
import os

from equos import (
    Equos,
    EquosAgent,
    CreateEquosAgentRequest,
    AgentProvider,
    GeminiRealtimeVoices,
    GeminiRealtimeModels,
)

from dotenv import load_dotenv

load_dotenv()


def create_agent() -> None:
    client = Equos(api_key=os.getenv("EQUOS_API_KEY", ""))

    agent: EquosAgent = client.agents.create(
        data=CreateEquosAgentRequest(
            provider=AgentProvider.gemini,
            name="English Translator",
            model=GeminiRealtimeModels.gemini_2_5_flash_native_audio_09_2025,
            voice=GeminiRealtimeVoices.Fenrir,
            instructions="You are a helpful assistant that translates English to French.",
            search=True,
            emotions=True,
            client=None,  # Optional: specify a client ID if needed for resource segmentation.
        )
    )

    print("Agent created:", agent.id)


if __name__ == "__main__":
    create_agent()
